sdk/loadtestservice/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/LoadTestRunAsyncClient.java

```diff
@@ -0,0 +1,1256 @@
+                                    status = testRun.get("status").asText();
+                                } catch (JsonProcessingException e) {
+                                    return Mono.error(
+                                            new RuntimeException(
+                                                    "Encountered exception while retriving test run status"));
+                                }
+                                LongRunningOperationStatus lroStatus;
+                                switch (status) {
+                                    case "NOTSTARTED":
+                                        lroStatus = LongRunningOperationStatus.NOT_STARTED;
+                                        break;
+                                    case "DONE":
+                                        lroStatus = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                                        break;
+                                    case "FAILED":
+                                        lroStatus = LongRunningOperationStatus.FAILED;
+                                        break;
+                                    case "CANCELLED":
+                                        lroStatus = LongRunningOperationStatus.USER_CANCELLED;
+                                        break;
+                                    default:
+                                        lroStatus = LongRunningOperationStatus.IN_PROGRESS;
+                                        break;
+                                }
+                                return Mono.just(new PollResponse<>(lroStatus, testRunBinary));
+                            });
+                },
+                (activationResponse, context) ->
+                        stopTestRunWithResponse(testRunId, defaultRequestOptions).flatMap(FluxUtil::toMono),
+                (context) -> getTestRunWithResponse(testRunId, defaultRequestOptions).flatMap(FluxUtil::toMono));
+    }
+
+    /**
+     * List the metric values for a load test run.
+     *
+     * <p><strong>Query Parameters</strong>
+     *
+     * <table border="1">
+     *     <caption>Query Parameters</caption>
+     *     <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     *     <tr><td>aggregation</td><td>String</td><td>No</td><td>The aggregation</td></tr>
+     *     <tr><td>interval</td><td>String</td><td>No</td><td>The interval (i.e. timegrain) of the query. Allowed values: "PT5S", "PT10S", "PT1M", "PT5M", "PT1H".</td></tr>
+     * </table>
+     *
+     * You can add these to a request with {@link RequestOptions#addQueryParam}
+     *
+     * <p><strong>Header Parameters</strong>
+     *
+     * <table border="1">
+     *     <caption>Header Parameters</caption>
+     *     <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     *     <tr><td>Content-Type</td><td>String</td><td>No</td><td>The content type. Allowed values: "application/json".</td></tr>
```

Comment:
@weidongxu-microsoft @srnagar Since we require user to pass Content-Type header for optional body, we should change the doc to reflect. Does this change requires removal for `@Generated` for it to persist across autorest regeneration

Reply:
But it is not required, as body is not required. If user does not provide body, user is not required to provide content-type.

Reply:
Yeah, but backend returns 415 if this header is not set due to .NET 6 limitations. Hence, we modified the Metrics API to add the header manually after removing `@Generated` annotation. What is the recommended approach for this?

Reply:
You mean even if user does not provide the body, backend will still fail with 415 when no content-type is provided?

Eh, I guess in this case customize the API is the solution.

Reply:
Yep, that's right. Although its changed in .NET 7 but we are yet to upgrade